From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Wahbe <ian@wahbe.com>
Date: Thu, 9 Mar 2023 18:22:25 +0100
Subject: [PATCH] Insert shim


diff --git a/shim/shim.go b/shim/shim.go
new file mode 100644
index 00000000..ac430274
--- /dev/null
+++ b/shim/shim.go
@@ -0,0 +1,10 @@
+package shim
+
+import (
+	"github.com/confluentinc/terraform-provider-confluent/internal/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func NewProvider(version, customUserAgent string) *schema.Provider {
+	return provider.New(version, customUserAgent)()
+}
