From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: guineveresaenger <guinevere@pulumi.com>
Date: Fri, 16 Aug 2024 15:50:38 -0700
Subject: [PATCH] Add Go module version to upstream go.mod


diff --git a/go.mod b/go.mod
index 41618299..1dff7b93 100644
--- a/go.mod
+++ b/go.mod
@@ -1,4 +1,4 @@
-module github.com/confluentinc/terraform-provider-confluent
+module github.com/confluentinc/terraform-provider-confluent/v2
 
 go 1.22.7
 
